modules/nf-scil/image/resample/main.nf

```diff
@@ -0,0 +1,60 @@
+process IMAGE_RESAMPLE {
+    tag "$meta.id"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://scil.usherbrooke.ca/containers/scilus_1.6.0.sif':
+        'scilus/scilus:1.6.0' }"
+
+    input:
+    tuple val(meta), path(image)
+
+    output:
+    tuple val(meta), path("*_resampled.nii.gz"), emit: image
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def ref = task.ext.ref ? "--ref" + task.ext.ref : ""
```

Comment:
I had a thought about this. Since the reference is a file on disk and the module needs to stage it at execution, we need to pass it as an **optional input** instead of through task.ext.

Reply:
Added ref as optional input as described in doc (path(f1) /* optional, value = [] */), but testing seems not to accept it as optional. Also included boolean logic to manage resampling method similarly to scil_volume_resample.py.

Reply:
I address this in my new review